test/sharness/t0062-daemon-api.sh

```diff
@@ -10,64 +10,64 @@ test_description="Test daemon command"
 test_init_ipfs
 
 differentport=$((PORT_API + 1))
-api_different="/ip4/127.0.0.1/tcp/$differentport"
+api_other="/ip4/127.0.0.1/tcp/$differentport"
 api_unreachable="/ip4/127.0.0.1/tcp/1"
 
 test_expect_success "config setup" '
-	api_fromcfg=$(ipfs config Addresses.API) &&
 	peerid=$(ipfs config Identity.PeerID) &&
 	test_check_peerid "$peerid"
 '
 
 test_client() {
-	printf "$peerid" >expected &&
-	ipfs "$@" id -f="<id>" >actual &&
-	test_cmp expected actual
+	opts="$@"
+	echo "OPTS = " $opts
+	test_expect_success "client must work properly $state" '
+		printf "$peerid" >expected &&
+		ipfs id -f="<id>" $opts >actual &&
+		test_cmp expected actual
+	'
 }
 
 test_client_must_fail() {
-	echo "Error: api not running" >expected_err &&
-	test_must_fail ipfs "$@" id -f="<id>" >actual 2>actual_err &&
-	test_cmp expected_err actual_err
+	opts="$@"
+	echo "OPTS = " $opts
+	test_expect_success "client should fail $state" '
+		echo "Error: api not running" >expected_err &&
+		test_must_fail ipfs id -f="<id>" $opts >actual 2>actual_err &&
+		test_cmp expected_err actual_err
+	'
 }
 
 test_client_suite() {
     state="$1"
     cfg_success="$2"
     diff_success="$3"
+	api_fromcfg="$4"
+	api_different="$5"
+
     # must always work
-    test_expect_success "client should work $state" '
-        test_client
-    '
+    test_client
 
     # must always err
-    test_expect_success "client --api unreachable should err $state" '
-	    test_client_must_fail --api "$api_unreachable"
-    '
```

Comment:
@jbenet when these would fail, we wouldnt get much indication to what actually failed, moving the 'test_expect' calls down lower makes debugging easier, instead of seeing that 'test_client' failed, we see that
```
		printf "$peerid" >expected &&
		ipfs id -f="<id>" $opts >actual &&
		test_cmp expected actual
```
failed